### ISSUE TITLE
Checkpoint

### ISSUE THREAD

**Issue by NSTAdventure** [00-Evan/shattered-pixel-dungeon#885]
hi dev! 👋😊

can you add checkpoint system or mechanism?

I don't want Restart from beginning if i killed :-(
that would be very annoying 🙏😔

edit:
bruh it's disliked why?

**Comment by Block-Vader** [comment]
1) This place is for bug reports, not suggestion. 
2) The developer said NO to this feature multiple times.

**Comment by NSTAdventure** [comment]
> 1. This place is for bug reports, not suggestion.
> 2. The developer said NO to this feature multiple times.

wait then what is this mean? because dev said that are welcome
(or did i misunderstood something?)
![Screenshot_2021-12-19-21-08-02-06_1](https://user-images.githubusercontent.com/50813528/146677873-5db40d92-1ba4-4a10-a97a-c53aca03315d.jpg)


(me be like: Bruh 🗿🗿🗿🗿🗿🗿)

**Comment by fregkos** [comment]
Please see #743.
tl;dr
This game is in roguelikes category.

**Comment by NSTAdventure** [comment]
> Please see #743. tl;dr This game is in roguelikes category.

okay sir 🙏
alternative why not add like practice mode or something else like that? or if dev doing this game no longer like rogue?

**Comment by fregkos** [comment]
> > Please see #743. tl;dr This game is in roguelikes category.
> 
> okay sir 🙏 alternative why not add like practice mode or something else like that? or if dev doing this game no longer like rogue?

Sir? I feel like a knight now, thanks. 😅

I'm not sure if I understand ypur question, but your answers are on the Evan's reply, on the issue I linked. He explains why saving states are not going to happen and why a "practice" mode creates bad mannerisms for the gameplay experience.

Please, make sure you read and understand it before you ask again, so we won't unnecessarily repeat ourselves. This question has been answered multiple times.

**Comment by NSTAdventure** [comment]
> > > Please see #743. tl;dr This game is in roguelikes category.
> > 
> > 
> > okay sir 🙏 alternative why not add like practice mode or something else like that? or if dev doing this game no longer like rogue?
> 
> Sir? I feel like a knight now, thanks. 😅
> 
> I'm not sure if I understand your question, but your answers are on the Evan's reply, on the issue I linked. He explains why saving states are not going to happen and why a "practice" mode creates bad mannerisms for the gameplay experience.
> 
> Please, make sure you read and understand it before you ask again, so we won't unnecessarily repeat ourselves. This question has been answered multiple times.

ok my last request is can you undo the dislike button on main thread above? i literally didn't know that will defeat roguelike games 🙏

**Comment by fregkos** [comment]
> > > > Please see #743. tl;dr This game is in roguelikes category.
> > > 
> > > 
> > > okay sir 🙏 alternative why not add like practice mode or something else like that? or if dev doing this game no longer like rogue?
> > 
> > 
> > Sir? I feel like a knight now, thanks. 😅
> > I'm not sure if I understand your question, but your answers are on the Evan's reply, on the issue I linked. He explains why saving states are not going to happen and why a "practice" mode creates bad mannerisms for the gameplay experience.
> > Please, make sure you read and understand it before you ask again, so we won't unnecessarily repeat ourselves. This question has been answered multiple times.
> 
> ok my last request is can you undo the dislike button on main thread above? i literally didn't know that will defeat roguelike games 🙏

That wasn't me. Don't pay attention to these small things.

I understand your confusion though. Your thoughts were the same as mine the first time I played any roguelike. Roguelikes teach you that your choices, big or small, have consequences and that you should learn through trial and error, the hard way. Having a way to practice without consequences does not teach you the best lessons. That's roughly the philosophy of Roguelikes.
